Make #19061 backward-compatible

When running the new `release/build-official-release.sh` against `release-1.1`, I get:

```
make: Entering directory `/tmp/kubernetes-build-release-v1.1.5-cmQJYh7/cluster/images/hyperkube'
make: *** No rule to make target `build'.  Stop.
make: Leaving directory `/tmp/kubernetes-build-release-v1.1.5-cmQJYh7/cluster/images/hyperkube'
```

The `build` make target didn't exist in `v1.1`, so that's a problem:

- [ ] Add a comment that `make release` is the *de facto* way to build a release, and one shouldn't add anything else
- [ ] Figure out a good workaround

xref #19061.

#19935 didn't work; the release process is still broken.

```
make: Entering directory `/tmp/kubernetes-build-release-v1.1.6-6cNPWJ1/cluster/images/hyperkube'
cp ./* /tmp/tmp.SdoPPOPQ04
cp ../../saltbase/salt/helpers/safe_format_and_mount /tmp/tmp.SdoPPOPQ04
cp ../../saltbase/salt/generate-cert/make-ca-cert.sh /tmp/tmp.SdoPPOPQ04
cp ../../../_output/dockerized/bin/linux/amd64/hyperkube /tmp/tmp.SdoPPOPQ04
cd /tmp/tmp.SdoPPOPQ04 && sed -i "s/VERSION/v1.1.6/g" master-multi.json master.json kube-proxy.json
sed: can't read kube-proxy.json: No such file or directory
make: *** [build] Error 2
make: Leaving directory `/tmp/kubernetes-build-release-v1.1.6-6cNPWJ1/cluster/images/hyperkube'
```

Bumping this back to P0.

--- a/build/common.sh
+++ b/build/common.sh
@@ -667,6 +667,13 @@ function kube::release::clean_cruft() {
   find ${RELEASE_STAGE} -name '.DS*' -exec rm {} \;
 }
 
+function kube::release::package_hyperkube() {
+  # If we have these variables set then we want to build all docker images.
+  if [[ -n "${KUBE_DOCKER_IMAGE_TAG-}" && -n "${KUBE_DOCKER_REGISTRY-}" ]]; then
+    REGISTRY="${KUBE_DOCKER_REGISTRY}" VERSION="${KUBE_DOCKER_IMAGE_TAG}" make -C cluster/images/hyperkube/ build
+  fi
+}
+
 function kube::release::package_tarballs() {
   # Clean out any old releases
   rm -rf "${RELEASE_DIR}"

--- a/build/release.sh
+++ b/build/release.sh
@@ -39,3 +39,4 @@ fi
 
 kube::build::copy_output
 kube::release::package_tarballs
+kube::release::package_hyperkube

--- a/release/build-official-release.sh
+++ b/release/build-official-release.sh
@@ -91,8 +91,6 @@ export KUBE_DOCKER_REGISTRY="gcr.io/google_containers"
 export KUBE_DOCKER_IMAGE_TAG="${KUBE_RELEASE_VERSION}"
 
 make release
-# We don't want to include this in 'make release' as it'd slow down every day development cycle.
-REGISTRY="${KUBE_DOCKER_REGISTRY}" VERSION="${KUBE_DOCKER_IMAGE_TAG}" make -C cluster/images/hyperkube/ build
 
 if ${KUBE_BUILD_DIR}/cluster/kubectl.sh version | grep Client | grep dirty; then
   echo "!!! Tag at invalid point, or something else is bad. Build is dirty. Don't push this build." >&2

